src/kernel/mem/slab.c

```diff
@@ -292,7 +295,7 @@ void _slab_free(struct slab *slab, const void *obj) {
   slab->freelist[i].stack_item = slab->freelist[slab->allocated].stack_item;
   slab->freelist[slab->allocated].stack_item = index;
 
-  slab->freelist[i].pos_in_stk = slab->allocated;
+  slab->freelist[index].pos_in_stk = slab->allocated;
```

Comment:
bugfix here

took way too long to debug. But better here than later.

This is a bit of an implementation detail, so it's a bit hard to debug. I may try to come up with a test case that would detect this